Packs/ApiModules/Scripts/FireEyeApiModule/FireEyeApiModule.py

```diff
@@ -0,0 +1,222 @@
+from CommonServerPython import *
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' CONSTANTS '''
+FE_DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+
+class FireEyeClient(BaseClient):
+    def __init__(self, base_url: str, username: str, password: str, verify: bool, proxy: bool):
+        super().__init__(base_url=base_url, auth=(username, password), verify=verify, proxy=proxy)
+        self._headers = {
+            'X-FeApi-Token': self._generate_token(),
+            'Accept': 'application/json',
+        }
+
+    @logger
+    def _generate_token(self) -> str:
+        resp = self._http_request(method='POST', url_suffix='auth/login', resp_type='response')
+        if resp.status_code != 200:
+            raise DemistoException(
+                f'Token request failed with status code {resp.status_code}. message: {str(resp)}')
+        return resp.headers['X-FeApi-Token']
```

Comment:
consider adding check here that the key `X-FeApi-Token` really exist and add raise an exception if not.